topostats/grainstats.py

```diff
@@ -300,10 +313,10 @@ def calculate_radius_stats(self, edges: list, points: list) -> Tuple:
         # Calculate the radius of each point
         radii = self._calculate_radius(displacements)
         return {
-            "min": np.nanmin(radii),
-            "max": np.nanmax(radii),
-            "mean": np.nanmean(radii),
-            "median": np.nanmedian(radii),
+            "min": np.min(radii),
```

Comment:
Okay so this might be the wrong way to go about it, but using `np.nanmedian()` etc would make it so the program doesn't crash when it comes across `radii` arrays that contain `nan`s. If a `radii` array contains a `nan`, then something has gone very wrong and therefore data from this shouldn't be trusted, so I let it fail here.

Reply:
Ok, that makes sense as its prefectly rationale.

With an eye on processing files in batches though this block should perhaps be encapsulated in a `try: ... except:` statement so that if one image fails others in the batch don't. Currently I've taken a fairly brute force approach to capturing this at a high level of wrapping the whole `grainstats` and `dnatracing` steps in such a `try: ... except:` but having finer grained exceptions that capture specific steps is preferable.

Might want to save this as an issue for the next Milestone though.

Reply:
Ok, that makes sense, its perfectly rationale.

With an eye on processing files in batches though this block should perhaps be encapsulated in a `try: ... except:` statement so that if one image fails others in the batch don't. Currently I've taken a fairly brute force approach to capturing this at a high level of wrapping the whole grainstats and dnatracing steps in such a `try: ... except:` but having finer grained exceptions that capture specific steps is preferable.

Might want to save this as an issue for the next Milestone though.